### ISSUE TITLE
JSDOM-react adapter should implement isDisplayed

### ISSUE THREAD

**Issue by NitayRabi** [wix-incubator/unidriver#52]
Currently `isDisplayed` implementation under JSDOM adapter returns `true`.
[Source](https://github.com/wix-incubator/unidriver/blob/5ea38eca0a97b5240f74fa43839d7239baeaa10c/adapters/jsdom-react/src/index.ts#L206)
Method should test element visibility similar to - 

[Protractor](https://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.isDisplayed)

(e.g Test existence, size over 0 X 0 and display !== 'none')

**Comment by GabiGrin** [comment]
I agree it shouldn't return `true`
the problem with jsdom is that you don't have proper measuring so you can't know

thus, I prefer to either:
1. throw an error
2. always return `false`

Thoughts? any other options we can do in JSDom?
@borisd9  @hepiyellow

**Comment by hepiyellow** [comment]
1. Certainly throwing an Unsupported Error is better than returning false. 
2. I this ,again, that jsdom is a special case. And we should have both a React adapter and a React/JSDom adapter. 
  2.a after such separation, i think jsdom adapter is the only adapter that we should allow to throw Unsupported Error.

**Comment by GabiGrin** [comment]
@hepiyellow what does it mean a "react" adapter alone? what environment does it run on?

**Comment by dor-itzhaki** [comment]
what about implementing this https://github.com/testing-library/jest-dom#tobevisible ?

**Comment by dor-itzhaki** [comment]
or just something like:
```
function isVisible (ele) {
    var style = window.getComputedStyle(ele);
    return  style.width !== "0" &&
    style.height !== "0" &&
    style.opacity !== "0" &&
    style.display!=='none' &&
    style.visibility!== 'hidden';
}
```

**Comment by GabiGrin** [comment]
@dor-itzhaki height and width might be 0 on valid cases too, but I am OK with implementing the same logic the did on testing-library

**Comment by liorcode** [comment]
Any progress on that? It took me a while to realize how come my tests always fail :( just when I used the debugger, I found this weird behaviour.
I think throwing an exception is much better than this unexpected behaviour, or going with the testing-library logic

**Comment by GabiGrin** [comment]
@liorcode no, a PR should be fairly easy, the challenge will be coordinating this release with all users of WSR and ensuring nothing breaks unexpectedly

**Comment by liorcode** [comment]
hmm I see. did you consider using some flag+warning for the new behavior and then removing it in the next major version of WSR? this is what WSR usually do for breaking changes.
e.g.

```
isDisplayed(realCheck = false) {
  if (!realCheck) { 
    console.warn('Warning: using mocked implementation of 'isDisplayed' which always returns true. This will change in the next version. Pass "true" to really check')
  }
}
```

**Comment by GabiGrin** [comment]
@liorcode the warning is great meanwhile, can be added now.
As for the fix, I prefer following semver and release a new major of the react jsdom adapter and have WSR upgrade at will

@shlomitc  thoughts?

**Comment by shlomitc** [comment]
I went through our code and didn't see anywhere we're using the `isDisplayed()` method in our code, so probably this will not break us.
@liorcode - where did this happen to you?
Overall I'm ok with both warning and error, as long as it is clear and displayed well in the logs.

**Comment by liorcode** [comment]
It happened for me when I used my own drivers, not WSR. I did some tests to make sure some element was not displayed, and I was very surprised when it failed. took me a while to realize that isDisplayed returned `true` for no reason :)

**Comment by AlmogAdziashvili** [comment]
@GabiGrin 
According to jest-dom doc:

```
An element is visible if all the following conditions are met:

- it does not have its css property display set to none
- it does not have its css property visibility set to either hidden or collapse
- it does not have its css property opacity set to 0
- its parent element is also visible (and so on up to the top of the DOM tree)
```

what do you think?

**Comment by GabiGrin** [comment]
@AlmogAdziashvili  Sounds good
